libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -255,16 +381,45 @@ Builder& Builder::algorithm(Algorithm algo) noexcept {
 }
 
 TangentSpaceMesh* Builder::build() {
-    mMesh->mOutput->algorithm = selectAlgorithm(mMesh->mInput);
-    // "Working in progress. Not for use"
-    return NULL;
+    // Work in progress. Not for use.
+    Algorithm algo = selectAlgorithm(mMesh->mInput);
+    MethodPtr method = nullptr;
+    switch (algo) {
+        case Algorithm::FRISVAD:
+            method = [](const TangentSpaceMeshInput* input, TangentSpaceMeshOutput* output) {
+                normalsOnlyMethod(input, output, frisvadKernel);
+            };
+            break;
+        case Algorithm::HUGHES_MOLLER:
+            method = [](const TangentSpaceMeshInput* input, TangentSpaceMeshOutput* output) {
+                normalsOnlyMethod(input, output, hughesMollerKernel);
+            };
+            break;
+        case Algorithm::FLAT_SHADING:
+            method = flatShadingMethod;
+            break;
+        default:
+            break;
+    }
+    assert_invariant(method);
+    method(mMesh->mInput, mMesh->mOutput);
+    return mMesh;
```

Comment:
I think this design is a bit problematic because a Builder can only be used once. e.g. this wouldn't work:

```
Builder builder;
builder.triangles();
auto a = builder.build();
auto b = builder.build();
```

So I think at least, we need to reset Builder to its initial state after `build()` and document it.

In addition, I don't like that *we* are allocating the mesh but we let the user destroy it. Instead, you should add a `static TangentSpaceMesh::destroy(TangentSpaceMesh*)` and make the dtor private.
This will allow us to use pools in the future for instance, and it'll protect us from binary compatibility issues too.

Reply:
done